packages/node/src/starknet/api.starknet.ts

```diff
@@ -162,22 +157,26 @@ export class StarknetApi implements ApiWrapper {
 
   /***
    * Get the latest block (with its header)
-   * we will try to get the latest block, if it is rejected, we will get the latest accepted block
    * @returns {Promise<BLOCK_HEADER>}
    */
-  async getFinalizedBlock(): Promise<SPEC.BLOCK_HEADER> {
+  async getFinalizedBlock(): Promise<Partial<SPEC.BLOCK_HEADER>> {
+    // we can not direct use this.client.getBlockLatestAccepted(), because its return missing parent block hash，but we still can retrieve the block number.
+    // In the meantime we also fetch the latest block, if it is rejected, we will get the latest accepted and fetch more details
     const [latestBlock, latestAccepted] = await Promise.all([
       this.client.getBlock('latest'),
-      await this.getFinalizedBlockHeight(),
+      this.getFinalizedBlockHeight(),
     ]);
-    // @ts-ignore
-    let b: SPEC.BLOCK_HEADER = latestBlock;
+
+    // If latest block been reject, we need to fetch the latest accepted block
     if (latestBlock.status === 'REJECTED') {
```

Comment:
I think this also needs to consider [`PENDING`](https://playground.open-rpc.org/?uiSchema%5BappBar%5D%5Bui:splitView%5D=false&schemaUrl=https://raw.githubusercontent.com/starkware-libs/starknet-specs/master/api/starknet_api_openrpc.json&uiSchema%5BappBar%5D%5Bui:input%5D=false&uiSchema%5BappBar%5D%5Bui:darkMode%5D=true&uiSchema%5BappBar%5D%5Bui:examplesDropdown%5D=false) as well. With that in mind, how frequent does this case match? Would it just be easier to call `getFinalizedBlockHeight` followed by `getBlockByHeightOrHash`?